File: src/main/java/net/yigitak/todoapp/models/Subtask.java
package net.yigitak.todoapp.models;


import lombok.*;
import org.bson.types.ObjectId;

import java.time.LocalDateTime;


@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode( of = { "id" } )
public class Subtask {

  private String id = new ObjectId().toHexString();

  private String  title;
  private String  description;
  private boolean completed = false;
  private LocalDateTime timeCreated = LocalDateTime.now();

}

File: src/main/java/net/yigitak/todoapp/models/Subtask.kt
package net.yigitak.todoapp.models

import org.bson.types.ObjectId
import java.time.LocalDateTime

data class Subtask(
    var id: String = ObjectId().toHexString(),

    var title: String? = null,
    var description: String? = null,
    var completed: Boolean = false,

    var timeCreated: LocalDateTime = LocalDateTime.now()
)
